File: Actividad 3/Actividad 3.1/4_3.py
class Estudiante:
    def __init__(self, nombres, apellidos, colegio, edad):
        self.nombres = nombres
        self.apellidos = apellidos
        self.colegio = colegio
        self.edad = edad

class Perro:
    def __init__(self, nombre, raza, edad):
        self.nombre = nombre
        self.raza = raza
        self.edad = edad

def menu_principal():
    print("Menu")
    print("1. Generar estudiante")
    print("2. Manejar estudiante")
    print("3. Salir")
    return int(input("Elija una opción: "))

def menu_estudiante():
    print("Menu")
    print("1. Crear objetos")
    print("2. Comparar objetos")
    print("3. Salir")
    return int(input("Elija una opción: "))

estudiantes = []
perros = []

while True:
    opcion = menu_principal()
    if opcion == 1:
        nombres = input("Ingrese nombres: ")
        apellidos = input("Ingrese apellidos: ")
        colegio = input("Ingrese colegio: ")
        edad = int(input("Ingrese edad: "))
        estudiantes.append(Estudiante(nombres, apellidos, colegio, edad))
    elif opcion == 2:
        nombres = input("Ingrese el nombre del estudiante: ")
        for estudiante in estudiantes:
            if estudiante.nombres == nombres:
                while True:
                    opcion = menu_estudiante()
                    if opcion == 1:
                        nombre = input("Ingrese nombre del perro: ")
                        raza = input("Ingrese raza del perro: ")
                        edad = int(input("Ingrese edad del perro: "))
                        perros.append(Perro(nombre, raza, edad))
                    elif opcion == 2:
                        if len(perros) >= 2:
                            nombre_perro1 = input("Ingrese el nombre del primer perro:")
                            nombre_perro2 = input("Ingrese el nombre del segundo perro:")
                            for perro1 in perros:
                                if perro1.nombre == nombre_perro1:
                                    for perro2 in perros:
                                        if perro2.nombre == nombre_perro2:
                                            if perro1.raza == perro2.raza:
                                                if perro1.edad == perro2.edad:
                                                    print("Los perros son los mismos")
                                                else:
                                                    print("Los perros son diferentes")
                                                break
                                    break
                            else:
                                print("No hay suficientes perros")
                        elif opcion == 3:
                            print("Saliendo...")
                            break
    elif opcion == 3:
        print("Saliendo...")
        break

File: Actividad 3/Actividad 3.1/4_3.java
import java.util.ArrayList;
import java.util.Scanner;

class Estudiante {
    String nombres;
    String apellidos;
    String colegio;
    int edad;

    public Estudiante(String nombres, String apellidos, String colegio, int edad) {
        this.nombres = nombres;
        this.apellidos = apellidos;
        this.colegio = colegio;
        this.edad = edad;
    }
}

class Perro {
    String nombre;
    String raza;
    int edad;

    public Perro(String nombre, String raza, int edad) {
        this.nombre = nombre;
        this.raza = raza;
        this.edad = edad;
    }
}

public class Main {
    static Scanner scanner = new Scanner(System.in);
    static ArrayList<Estudiante> estudiantes = new ArrayList<>();
    static ArrayList<Perro> perros = new ArrayList<>();

    public static void main(String[] args) {
        while (true) {
            int opcion = menuPrincipal();
            if (opcion == 1) {
                generarEstudiante();
            } else if (opcion == 2) {
                manejarEstudiante();
            } else if (opcion == 3) {
                System.out.println("Saliendo...");
                break;
            }
        }
    }

    static int menuPrincipal() {
        System.out.println("Menu");
        System.out.println("1. Generar estudiante");
        System.out.println("2. Manejar estudiante");
        System.out.println("3. Salir");
        System.out.print("Elija una opción: ");
        return scanner.nextInt();
    }

    static void generarEstudiante() {
        System.out.print("Ingrese nombres: ");
        String nombres = scanner.next();
        System.out.print("Ingrese apellidos: ");
        String apellidos = scanner.next();
        System.out.print("Ingrese colegio: ");
        String colegio = scanner.next();
        System.out.print("Ingrese edad: ");
        int edad = scanner.nextInt();
        estudiantes.add(new Estudiante(nombres, apellidos, colegio, edad));
    }

    static void manejarEstudiante() {
        System.out.print("Ingrese el nombre del estudiante: ");
        String nombres = scanner.next();
        for (Estudiante estudiante : estudiantes) {
            if (estudiante.nombres.equals(nombres)) {
                while (true) {
                    int opcion = menuEstudiante();
                    if (opcion == 1) {
                        crearPerro();
                    } else if (opcion == 2) {
                        compararPerros();
                    } else if (opcion == 3) {
                        System.out.println("Saliendo...");
                        break;
                    }
                }
            } else {
                System.out.println("Estudiante no encontrado");
            }
        }
    }

    static int menuEstudiante() {
        System.out.println("Menu");
        System.out.println("1. Crear objetos");
        System.out.println("2. Comparar objetos");
        System.out.println("3. Salir");
        System.out.print("Elija una opción: ");
        return scanner.nextInt();
    }

    static void crearPerro() {
        System.out.print("Ingrese nombre del perro: ");
        String nombre = scanner.next();
        System.out.print("Ingrese raza del perro: ");
        String raza = scanner.next();
        System.out.print("Ingrese edad del perro: ");
        int edad = scanner.nextInt();
        perros.add(new Perro(nombre, raza, edad));
    }

    static void compararPerros() {
        if (perros.size() >= 2) {
            System.out.print("Ingrese el nombre del primer perro:");
            String nombrePerro1 = scanner.next();
            System.out.print("Ingrese el nombre del segundo perro:");
            String nombrePerro2 = scanner.next();
            for (Perro perro1 : perros) {
                if (perro1.nombre.equals(nombrePerro1)) {
                    for (Perro perro2 : perros) {
                        if (perro2.nombre.equals(nombrePerro2)) {
                            if (perro1.raza.equals(perro2.raza) && perro1.edad == perro2.edad) {
                                System.out.println("Los perros son los mismos");
                            } else {
                                System.out.println("Los perros son diferentes");
                            }
                            return;
                        }
                    }
                }
            }
            System.out.println("No se encontraron los perros");
        } else {
            System.out.println("No hay suficientes perros");
        }
    }
}
